### ISSUE TITLE
docker-desktop stuck on "Docker Desktop stopped" ubuntu

### ISSUE THREAD

**Issue by bro-aero** [docker/desktop-linux#52]
- [x] I have tried with the latest version of Docker Desktop
  - [ ] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 595b6919-652b-40ce-9c12-d496e5b896a7/20220724231742

### Expected behavior
To be honest, as I'm new to docker desktop I don't know what to expect specifically, but I'm assuming I should have options to start new containers etc.

### Actual behavior
docker-desktop displays the message "Docker Desktop stopped" and it seems like most options/actions are unusable.
![image](https://user-images.githubusercontent.com/109927345/180670592-6b74af74-dc9b-43ba-a039-f070da7ec460.png)

### Information
This machine is running Ubuntu 20.04.4 LTS and it's not a VM
Note: I am willing to try any suggestions that may help, however I have already tried:
- adding my user to kvm group
- adding my user to docker group
- running docker-desktop with docker-engine enabled/running
- running docker-desktop with docker-engine disabled
- restarting my machine
- resetting docker-desktop to defaults
- clearing docker-desktop cache
-reinstalling docker-desktop

Also note:
when installing docker-desktop I have been downloading docker-desktop-4.10.1-amd64.deb and installing using the ubuntu software manager.

I'm able to reproduce this by opening docker-desktop, however I'm assuming this could be caused by some conflict with something that is specific to my machine, so as for re-creating an environment, I'm not sure.

Thanks in advance to anybody that has suggestions or can help.

Let me know if I have missed any details or info, happy to add more.

**Comment by aiordache** [comment]
Hi @bro-aero. The logs in the diagnostics bundle you uploaded point to a kvm setup issue:
```
Could not access KVM kernel module: No such file or directory
qemu-system-x86_64: failed to initialize KVM: No such file or directory
```
Docker Desktop is running a VM and for that it needs kvm virtualization support. 
Please follow the instructions in the [install docs](https://docs.docker.com/desktop/install/linux-install/#kvm-virtualization-support) to enable the module and set up device access.

**Comment by bro-aero** [comment]
Right, yes makes sense, thanks for pointing that out.

I thought I had followed that step during installation, but looking at it now it appears my CPU does not support KVM.

Any chance I can use qemu somehow rather than kvm?

**Comment by p1-0tr** [comment]
@bro-aero - sorry, we don't have an option to run without kvm. And adding it would have a massive performance penalty. Did you make sure you have enabled virtualisation in the BIOS (https://bce.berkeley.edu/enabling-virtualization-in-your-pc-bios.html)?

**Comment by CaCaBlocker** [comment]
Hi, @bro-aero 
How are you doing? I got the same issue, and I didn't fix it yet. I hope your help.

**Comment by Marhamat92** [comment]
Not any solution? I still have same problem,please help me guys :(

**Comment by Tarrowren** [comment]
> Hi, @bro-aero How are you doing? I got the same issue, and I didn't fix it yet. According to @aiordache's advice, I checked the kvm setup, and I found a problem. ![image](https://user-images.githubusercontent.com/87882515/188402236-b510fdd0-ca93-4d87-ad9d-542569ef4172.png) ![image](https://user-images.githubusercontent.com/87882515/188402113-3776f8a9-d7cb-46e1-b763-5276756f1f18.png) I logged out, and then logged in, and I saw my user didn't have permssion. ![image](https://user-images.githubusercontent.com/87882515/188402599-51303811-e229-4ef7-91cf-819d8aac4ce0.png)

This is to add you to the kvm group, just use `groups` to check

--------------
I checked kvm and everything is fine, but I still can't start docker desktop.

And I was able to start it normally at first, the problem appeared after a reboot of the computer, when it was solved after restarting the application, but now I can't fix it no matter how much I restart

**Comment by Marhamat92** [comment]
Bro I checked my pc because of this and working on laptop now ,it is working now ..maybe ı didnt do correct setup on first pc thats why...here installed it with steps that docker shows on documents for ubuntu

**Comment by CaCaBlocker** [comment]
Hi, I already fixed my issue through this post. I'd appreciate if it's a bit of help for you to start docker.
https://stackoverflow.com/questions/72281976/docker-is-running-docker-desktop-says-docker-desktop-stopped

**Comment by BusinessOcean** [comment]
This command works for me and also chek for vm enabled in you bios

``systemctl --user start docker-desktop``

**Comment by pauljurczak** [comment]
I had the same problem. Here is the solution, which work for me:

1. Enabled Intel VMX in BIOS (AMD virtualization if you have AMD CPU)
2. Followed https://docs.docker.com/desktop/install/linux-install/#kvm-virtualization-support

**Comment by sujaykundu777** [comment]
For Other Laptops, please check the Bios settings and set Hyper-V or Virtualization to **Enabled** while starting your OS and save and exit.

**Comment by maliongan** [comment]
> I had the same problem. Here is the solution, which worked for me:
> 
>     1. Enabled Intel VMX in BIOS (AMD virtualization if you have AMD CPU)
> 
>     2. Followed https://docs.docker.com/desktop/install/linux-install/#kvm-virtualization-support

Thank you very much. Problem solved!!

**Comment by AIGeneratedUsername** [comment]
I got a `Docker Desktop stopped...` message after editing "Shared path" in its settings and clicking "Save and restart".

**Comment by cheack** [comment]
@AIGeneratedUsername 
> I got a `Docker Desktop stopped...` message just after editing "Shared path" in its settings and clicking "Save and restart". Very annoying that a simple settings editing may break Docker Desktop. Still don't know how to fix it.

I encountered the exact same problem, manually editing the config `/home/$USER/.docker/desktop/settings.json` helped.

```
  "filesharingDirectories": [
    "/snap/phpstorm/", <------ that was a problem. Removing this line helped.
    "/home"
  ],

```

**Comment by m1k439** [comment]
> @AIGeneratedUsername
> 
> > I got a `Docker Desktop stopped...` message just after editing "Shared path" in its settings and clicking "Save and restart". Very annoying that a simple settings editing may break Docker Desktop. Still don't know how to fix it.
> 
> I encountered the exact same problem, manually editing the config `/home/$USER/.docker/desktop/settings.json` helped.
> 
> ```
>   "filesharingDirectories": [
>     " /snap/phpstorm/", <------ that was a problem. Removing the leading space helped.
>     "/home"
>   ],
> ```
> 
> it's definitely a bug, сreated the issue - #122

Thanks for this ... I had a path to a drive that was no longer mounted -- removing it from `settings.json` allowed Docker Desktop to start

**Comment by cheack** [comment]
@m1k439 
Yeah, same for me, added another issue - https://github.com/docker/desktop-linux/issues/132

**Comment by ken-980** [comment]
Issue solved on unbuntu 23.04 on running on intel, had to enable virtualization in BIOS as said above, thanks
